scripts/blocks-utils.js

```diff
@@ -133,6 +148,85 @@ function fetchData(url, callback) {
     });
 }
 
+function formDataToJSON(formData) {
+  const jsonObject = {};
+  formData.forEach((value, key) => {
+    // eslint-disable-next-line no-prototype-builtins
+    if (!jsonObject.hasOwnProperty(key)) {
+      jsonObject[key] = value;
+    } else {
+      if (!Array.isArray(jsonObject[key])) {
+        jsonObject[key] = [jsonObject[key]];
+      }
+      jsonObject[key].push(value);
+    }
+  });
+  return JSON.stringify(jsonObject);
+}
+
+/**
+ * Posts form data to the given URL and calls the callback function with the response.
+ * @param url
+ * @param formData
+ * @param callback
+ * @param options
+ * @example
+ * const formData = new FormData();
+ * formData.append('apiName', 'getdata');
+ * postFormData('https://example.com/data', formData, (error, data) => {
+ *  if (error) {
+ *  console.error('Error fetching data:', error);
+ *  }
+ *  else {
+ *  console.log('Data fetched successfully:', data);
+ *  }
+ *
+ */
+function postFormData(url, formData, callback, options = {}) {
+  const requestOptions = {
+    method: 'POST',
+    headers: options.headers || {},
+    body: formDataToJSON(formData),
```

Comment:
consumers can send the JSON object instead of form data directly, we can avoid this conversion